scripts/windows_dev_setup.ps1

```diff
@@ -0,0 +1,154 @@
+	}
+	else {
+		Write-Host "Unsupported Windows OS"
+		Exit
+	}
+}
+
+function check_package {
+  param(
+    [string]$package
+  )
+  if ($(winget list --name $package) -match 'No installed package found matching input criteria.') {
+    Write-Host "Installing $package..."
+    return $true
+  }   
+  else {
+    Write-Host "$package is already installed..."
+  }
+}
+
+function install_msvc_build_tools {  # Installs C++ build tools, CMake, and Windows 10/11 SDK
+  $result = check_package "buildtools"
+  if ($result) {
+	install_variant
+	}
+}
+
+function install_variant {  # Decides between the Windows 10 SDK and Windows 11 SDK based on your OS
+	if ($global:os -eq "Windows 11") {
+		winget install Microsoft.VisualStudio.2022.BuildTools --accept-source-agreements --silent --override "--wait --quiet --add ProductLang En-us --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 --add Microsoft.VisualStudio.Component.Windows11SDK.22621 --add Microsoft.VisualStudio.Component.VC.CMake.Project --includeRecommended"
+		}
+	else {
+		winget install Microsoft.VisualStudio.2022.BuildTools --accept-source-agreements --silent --override "--wait --quiet --add ProductLang En-us --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 --add Microsoft.VisualStudio.Component.Windows10SDK.20348 --add Microsoft.VisualStudio.Component.VC.CMake.Project --includeRecommended"
+		}
+}
+
+function get_msvc_version {  # Finds the MSVC version number and creates a valid filepath to add as an environment variable
+    $pathpattern = "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\*\bin\Hostx64\x64\link.exe"
+
+    # Get the file path that matches the pattern
+    $filepath = Get-ChildItem -Path $pathpattern -Recurse -ErrorAction SilentlyContinue | Select-Object -First 1
+
+    if ($filepath) {
+        # Extract the version number from the file path using regex
+        $global:msvcversion = $filepath.FullName -replace ".*MSVC\\(\d+\.\d+\.\d+)\\.*", '$1'
+    } else {
+        Write-Warning "MSVC not found: $pathpattern"
+        return $null
+    }
+}
+
+function verify_architecture {  # Checks whether the Windows machine is 32-bit or 64-bit
+	$result = Get-WmiObject -Class Win32_Processor | Select-Object AddressWidth
+	if ($result.Contains("64")) {
+		$global:architecture = "64"
+		}
+	else {
+		$global:architecture = "86"
+		}
+}
+
+function set_msvc_env_variables {  # Sets the environment variables based on the architecture and MSVC version
+	[Environment]::SetEnvironmentVariable("PATH", "$env:PATH;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\$global:msvcversion\bin\Hostx$global:architecture\x$global:architecture\link.exe", "User")
+	[Environment]::SetEnvironmentVariable("PATH", "$env:PATH;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\$global:msvcversion\bin\Hostx$global:architecture\x$global:architecture\cl.exe", "User")
+	Write-Host "Environment variables set"
+}

```

Comment:
I hope that someone doesn't install MSVC elsewhere, but maybe is there a way to find where MSVC is installed?  Like in linux it's `which msvc` or it would be something like that

Reply:
Thanks! Chocolatey seems like a great option as well, but I primarily chose WinGet since it's officially supported by Microsoft and has been included by default ever since 2018.

I'm unable to use `where` since the MSVC build tools installer doesn't set anything I can extract something useful from the PATH variable. I'm looking into using CIM PowerShell functionality to find where MSVC is installed. Worst case, I'll use Windows registry.

Reply:
@gregnazario I added a lot of stuff to the script. Please take another look when you get a chance. I'm satisfied with where it's at for V1 of the script. If there are no glaring issues, I'd like to get it published for now and update it later to include the operations tools in the future.